### DISCUSSION TITLE
TripsLayer continuous live updates without complete rerender

### DISCUSSION THREAD

**Discussion by Fabioni** [visgl/deck.gl#8872] [General]
Hey, I have a TripsLayer that show shows a trip, so far so good :D 

The trip is a actual live happening trip of a vehicle that sends its new position every second. I want to show the complete trip on the map. Currently, I think what happens is that the TripsLayer re-renders every second, however only one waypoint was added to the path array.

I thought the deckgl's "incremental append" would be the correct feature, but this only allows to append new paths, but not to add new waypoint to existing paths.

Also looking at this animation https://deck.gl/examples/trips-layer I felt like this is what I need, but this is actually kind of fake, because the complete trip is already known and only the variable `currentTime` changes and creates the animation.

**Comment by Pessimistress** [comment]
You do not have to append the new waypoints to their respective paths.

At t=0:

```js
data: [
  {id: 'vehicle-0', waypoints: [p00, p01, p02]},
  {id: 'vehicle-1', waypoints: [p10, p11]}
]
```

At t=1:

```js
data: [
  {id: 'vehicle-0', waypoints: [p00, p01, p02]},
  {id: 'vehicle-1', waypoints: [p10, p11]},
  {id: 'vehicle-0', waypoints: [p02, p03]},
  {id: 'vehicle-1', waypoints: [p11, p12]},
]
```

For the overlapping parts of the two segments to render correctly, you may need to adjust the blending [parameters](https://deck.gl/docs/api-reference/core/layer#parameters).

If your app is supposed to run for a long time, it is also a good idea to [start a new layer](https://deck.gl/docs/developer-guide/performance#handle-incremental-data-loading) from time to time and remove the layers that are too far in the past.

**Comment by Fabioni** [comment]
Thanks a lot!
I am not quite sure if I understand that correctly. Is the `id` field used for something or just for showcasing the idea?

**Reply by Pessimistress** [reply]
Just for showcasing the idea.
